prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -147,9 +153,19 @@ def maybe_initialize_agent(
             functions_config=FunctionsConfig.from_system_prompt_choice(
                 st.session_state.system_prompt_select
             ),
+            enable_langfuse=ENABLE_LANGFUSE,
+        )
+        # Use the un-observed version of these if Langfuse is enabled, because we don't want to have separate traces for these.
+        (
+            st.session_state.agent.reset()
+            if not ENABLE_LANGFUSE
```

Comment:
Do we not get the same behaviour in microchain as we have in pmat? If `ENABLE_LANGFUSE == False` then we don't do `agent.init_langfuse()`, so don't do `observe()(some_function)`, so this shouldn't be needed?

Reply:
In his library, I didn't put `@observe` on everything by default, but it's behind a flag: https://github.com/galatolofederico/microchain/blob/main/microchain/engine/agent.py#L33

Reply:
Maybe we don't need to track this streamlit application if it's bringing more complications than utility?

Reply:
Refactorer a bit, I introduced `total_iterations` state variable used as `resume` trigger:

```
agent.run(iterations=1, resume=st.session_state.total_iterations > 0)
```

so at the very first iteration, resume=False and `reset` + `build_initial_messages` are triggered inside of Microchain:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/4bf4f792-5af6-42d7-ac20-1693772f767f)